File: 24-adv-ds1/1-intro/main.cpp
// Advanced Data Structures 1 - Intro C++ example

#include <iostream>
#include <vector>

class UnionFind {
public:
    std::vector<int> parent, rank;
    UnionFind(int size) {
        parent.resize(size);
        rank.resize(size, 0);
        for(int i = 0; i < size; i++)
            parent[i] = i;
    }
};

int main() {
    UnionFind uf(5);
    std::cout << "Union-Find initialized for 5 elements." << std::endl;
    return 0;
}

File: 24-adv-ds1/1-intro/main.java
// Advanced Data Structures 1 - Intro Java example

class Main {
    static class UnionFind {
        int[] parent, rank;
        public UnionFind(int size) {
            parent = new int[size];
            rank = new int[size];
            for (int i = 0; i < size; i++) parent[i] = i;
        }
    }

    public static void main(String[] args) {
        UnionFind uf = new UnionFind(5);
        System.out.println("Union-Find initialized for 5 elements.");
    }
}
